From 10f1fb90ab8a364117b45fa2ad735622e9b5b8df Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 14 Nov 2016 19:09:18 +0200
Subject: [PATCH 0588/1345] dt-bindings: net: add phy-speed description to
 marvell-pp2x.txt bindings

commit  97bf3334f64f27811ca1a062fd9a79594f91a9a9 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- add phy-speed description to marvell-pp2x.txt bindings.
- update Example with phy-speed and relevant register address space.

Change-Id: Ia32e755748d55f0838b133e7610a9c9f93db6880
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33856
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/net/marvell-pp2x.txt       |   14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/Documentation/devicetree/bindings/net/marvell-pp2x.txt b/Documentation/devicetree/bindings/net/marvell-pp2x.txt
index 71c2904..a72114b 100644
--- a/Documentation/devicetree/bindings/net/marvell-pp2x.txt
+++ b/Documentation/devicetree/bindings/net/marvell-pp2x.txt
@@ -38,6 +38,7 @@ emac required properties:
 - interrupts: Link change interrupt for the mac.
 - port-id: Should be '0', '2', or '3' for ethernet macs, references the mac-id (GOP port-id).
 - phy-mode: See ethernet.txt file in the same directory
+- phy-speed: For 2.5Gbits SGMII port 2500 phy-speed should be set(used only for 2.5Gbits SGMII port)
 
 emac optional properties:
 ========================
@@ -52,15 +53,14 @@ gop {
 		mac-address = [00 00 00 00 00 01];
 		port-id = <0>; /* gop_port_id */
                 phy-mode = "sgmii"; /* lane-2 */
-                status = "okay";
         };
 	emac2: mac2 {
 		interrupts = <GIC_SPI 35 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 02];
 		port-id = <2>; /* gop_port_id */
-                phy-mode = "rgmii"; /* rgmii-0 */
+		phy-mode = "sgmii"; /* lane-0 */
+		phy-speed = <2500>;
                 phy = <&phy2>;
-                status = "okay";
 	};
 	emac3: mac3 {
 		interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
@@ -68,15 +68,13 @@ gop {
 		port-id = <3>; /* gop_port_id */
                 phy-mode = "rgmii"; /* rgmii-1 */
                 phy = <&phy3>;
-                status = "okay";
 	};
 };
 
 
 ppv22@000000 {
 	compatible = "marvell,mv-pp22";
-	reg = <0x000000 0x80000>, /* Packet Processor regs */
-	      <0x120000 0x9000>, /* SERDES regs */
+	reg = <0x000000 0x90000>, /* Packet Processor regs */
 	      <0x129000 0x0600>, /* XMIB regs */
 	      <0x12a000 0x200>,  /* LED regs */
 	      <0x12a200 0x200>,  /* SMI regs */
@@ -88,8 +86,8 @@ ppv22@000000 {
 	      <0x130600 0x200>,  /* FCA - flow control regs*/
 	      <0x130e00 0x100>,  /* MSPG - GMAC regs */
 	      <0x130f00 0x100>,  /* MSPG - XLG MAC regs */
-	      <0x441000 0x1000>;  /* RFU-1 Regs */
-	reg-names = "pp", "serdes", "xmib", "led", "smi", "tai", "xsmi", "mg", "mspg", "xpcs",
+	      <0x441100 0x100>;  /* RFU-1 Regs */
+	reg-names = "pp", "xmib", "led", "smi", "tai", "xsmi", "mg", "mspg", "xpcs",
 		    "fca", "gmac", "xlg", "rfu1";
 	clocks = <&gateclk 3>, <&gateclk 18>, <&gateclk 9>, <&gateclk 6>, <&gateclk 5>;
 	clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
-- 
1.7.9.5

